servers/src/main/java/tachyon/master/block/BlockMaster.java

```diff
@@ -459,6 +459,8 @@ public long getWorkerId(NetAddress workerNetAddress) {
         final long lostWorkerId = lostWorkerInfo.getId();
         LOG.warn("A lost worker {} has requested its old id {}.", workerAddress, lostWorkerId);
 
+        // Update the timestamp of the worker before it is considered an active worker.
+        lostWorkerInfo.updateLastUpdatedTimeMs();
```

Comment:
I added a unit test. I had to add a new method to MasterWorkerInfo() to set the last updated timestamp though.